pkg/archive/archive_test.go

```diff
@@ -207,6 +207,164 @@ func TestCmdStreamGood(t *testing.T) {
 	}
 }
 
+func TestUntarPathWithInvalidSrc(t *testing.T) {
+	_, err := os.Create("/tmp/dest")
+	if err != nil {
+		t.Fatalf("Fail to create the destination file")
+	}
+	defer os.RemoveAll("/tmp/dest")
+	err = UntarPath("", "/tmp/dest")
+	if err == nil {
+		t.Fatalf("UntarPath with invalid src path should throw an error.")
+	}
+}
+
+func TestUntarPath(t *testing.T) {
+	cmd := exec.Command("/bin/sh", "-c", "touch /tmp/src && tar cfT /tmp/src.tar /tmp/src")
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll("/tmp/src/")
+	defer os.RemoveAll("/tmp/src.tar")
+	_, err = os.Create("/tmp/dest")
+	if err != nil {
+		t.Fatalf("Fail to create the destination file")
+	}
+	defer os.RemoveAll("/tmp/dest")
+	err = UntarPath("/tmp/src.tar", "/tmp/dest")
+	if err != nil {
+		t.Fatalf("UntarPath shouldn't throw an error, %s.", err)
```

Comment:
I wonder why this shouldn't fail since dest isn't a dir - its file. Even if the tar is empty (like you created) I kind of expected things to blow up.